policy/pa.go

```diff
@@ -174,24 +174,24 @@ func isDNSCharacter(ch byte) bool {
 
 var (
 	errInvalidIdentifier    = berrors.MalformedError("Invalid identifier type")
-	errNonPublic            = berrors.MalformedError("Name does not end in a public suffix")
-	errICANNTLD             = berrors.MalformedError("Name is an ICANN TLD")
-	errPolicyForbidden      = berrors.RejectedIdentifierError("Policy forbids issuing for name")
-	errInvalidDNSCharacter  = berrors.MalformedError("Invalid character in DNS name")
-	errNameTooLong          = berrors.MalformedError("DNS name too long")
-	errIPAddress            = berrors.MalformedError("Issuance for IP addresses not supported")
-	errTooManyLabels        = berrors.MalformedError("DNS name has too many labels")
-	errEmptyName            = berrors.MalformedError("DNS name was empty")
-	errNameEndsInDot        = berrors.MalformedError("DNS name ends in a period")
-	errTooFewLabels         = berrors.MalformedError("DNS name does not have enough labels")
-	errLabelTooShort        = berrors.MalformedError("DNS label is too short")
-	errLabelTooLong         = berrors.MalformedError("DNS label is too long")
-	errMalformedIDN         = berrors.MalformedError("DNS label contains malformed punycode")
-	errInvalidRLDH          = berrors.RejectedIdentifierError("DNS name contains a R-LDH label")
-	errTooManyWildcards     = berrors.MalformedError("DNS name had more than one wildcard")
-	errMalformedWildcard    = berrors.MalformedError("DNS name had a malformed wildcard label")
-	errICANNTLDWildcard     = berrors.MalformedError("DNS name was a wildcard for an ICANN TLD")
-	errWildcardNotSupported = berrors.MalformedError("Wildcard names not supported")
+	errNonPublic            = berrors.MalformedError("Domain name does not end with a valid public suffix (TLD)")
+	errICANNTLD             = berrors.MalformedError("Domain name ends with a reserved ICANN TLD. Certificate can not be issued")
+	errPolicyForbidden      = berrors.RejectedIdentifierError("Policy forbids issuing a certificate for the domain name. Part or all of the name may be blacklisted")
+	errInvalidDNSCharacter  = berrors.MalformedError("Domain name contains invalid character")
+	errNameTooLong          = berrors.MalformedError("Domain name too long. (Note that certificates containing domain names over 230 octets long are not currently supported)")
+	errIPAddress            = berrors.MalformedError("Certificate issue for IP addresses is not supported")
+	errTooManyLabels        = berrors.MalformedError("Domain name has too many labels. Domain names can have up to 10 labels (parts)")
+	errEmptyName            = berrors.MalformedError("Domain name is empty")
+	errNameEndsInDot        = berrors.MalformedError("Domain name ends in a period")
+	errTooFewLabels         = berrors.MalformedError("Domain name does not have enough labels (parts). At least two parts are required for a valid domain name")
```

Comment:
```suggestion
	errTooFewLabels         = berrors.MalformedError("Domain name needs at least one dot")
```

Reply:
I think "needs at least one dot" clarifies pretty well. I think most people getting this error didn't try to issue for a TLD, they issued for some random string like "mycatisfluffy", or they typoed and left out the dot, like "examplecom".